crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -296,10 +321,14 @@ impl GenericCommandState {
             ".topic.operation" => self.operation().map(|s| s.into()),
             ".topic.cmd_id" => self.cmd_id().map(|s| s.into()),
             ".payload" => Some(self.payload.clone()),
-            path => path
-                .strip_prefix(".payload.")
-                .and_then(|path| json_excerpt(&self.payload, path))
-                .cloned(),
+            path if path.contains(['[', ']']) => None,
```

Comment:
I'm slightly confused as to what case this is covering? Is it explicitly filtering out TOML table headings? If so, why isn't `strip_prefix` covering this case?

Reply:
The point is to [exclude JSON path where  a bracket notation is used to access an item below an array](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_api/src/workflow/state.rs#L756). Such a path is currently considered as ill-formed (the path expression is left unchanged), by contrast with paths leading to an unknown value  (the path expression being then substituted for the empty string). See https://github.com/thin-edge/thin-edge.io/pull/2918#discussion_r1627076669.

Reply:
Why is accessing an array value considered invalid? Because array indexes positions are non-deterministic? Or to avoid out-of-bound index risks? If it was the latter, I'd have used the same logic for that as unknown path in an existing root: returning empty.